### ISSUE TITLE
Tone equalize doesn't like overexposed regions.

### ISSUE THREAD

**Issue by groutr** [darktable-org/darktable#3116]
**Describe the bug**
Using exposure module in combination with tone equalizer produces black spots on image in overexposed regions.

**To Reproduce**
1. Open photo
2. Raise exposure like +2EV (to make the effect more apparent)
3. Activate tone equalizer.  Black spots.

**Expected behavior**
I expected the bright parts to clip.  It looks like it wraps around to black.  Looking at the spectrograph of the image, it looks like the tone equalizer inverts the overexposed area.

**Screenshots**
[Before tone equalizer](https://postimg.cc/dkT9C2F3)

[After tone equalizer](https://postimg.cc/GBtKL0Sk)

**Platform (please complete the following information):**
 - OS: archlinux
 - Version: master: 1316a61e3051e6ecfe9c933d7533e7ace5a2f7ef

**Comment by dcendal** [comment]
Same issue

this is OK
![1](https://user-images.githubusercontent.com/48381321/66606725-052cb480-ebb3-11e9-8a3a-34baa9c6849a.png)
+1 filter difussion, now the indicator is -16 EV
![3](https://user-images.githubusercontent.com/48381321/66606724-052cb480-ebb3-11e9-978c-0c356d6a247c.png)
Any change applied in the parameters, only mask configuration
![2](https://user-images.githubusercontent.com/48381321/66606723-052cb480-ebb3-11e9-8c48-fe21a43fa00c.png)

Reverting d3bbd260cecf11fb0026fc7c79abaa4c63c2c887, I were not able to replicate

**Comment by cigno5** [comment]
I had the same issue on Ubuntu 18.4. 

I took one picture already overexposed (on raw data) and I enabled tone equalizer module, immediate result as groutr described

**Comment by groutr** [comment]
Same as @dcendal, reverting https://github.com/darktable-org/darktable/commit/d3bbd260cecf11fb0026fc7c79abaa4c63c2c887 seems to fix the problem.

**Comment by jenshannoschwalm** [comment]
This is related to masking.
You can either do "mask blending: off"  or adapt the "mask exposure compensation".

@aurelienpierre Maybe it's better to have mask blending off as default?

**Comment by ctouz** [comment]
Same issue on all the pictures  I had applied the tone equalizer before.
Ubuntu 19.04
![tone](https://user-images.githubusercontent.com/28384260/66658110-16bc9d80-ec41-11e9-9317-171a1e56df0c.png)

**Comment by groutr** [comment]
@jenshannoschwalm I tried your suggestions and they seem to work.  Thank you!

I guess I'm not entirely up to speed on how the masking works.

**Comment by aurelienpierre** [comment]
> Same as @dcendal, reverting d3bbd26 seems to fix the problem.

Actually, it does not fix the problem. The default mask exposure and contrast compensations have been tweaked around the bug fixed by #d3bbd26. Now that it's fixed, the mask exposure compensation need to be adjusted, and probably just set to 0.

Back to the original bug:

>  Open photo
    Raise exposure like +2EV (to make the effect more apparent)
    Activate tone equalizer. Black spots.

Use the mask exposure compensation to slide and spread the histogram in the valid area [-8; 0] EV. If you pushed the exposure (in exposure module) by 2EV, you would need to pull it by -2EV in mask compensation.

**Comment by aurelienpierre** [comment]
also @groutr :
https://www.youtube.com/watch?v=UuB9khJIrDI&lc=z22zzv44kpj2elt0xacdp435n0e5vz0wctjodskcrydw03c010c

**Comment by groutr** [comment]
@aurelienpierre, thanks!

**Comment by jenshannoschwalm** [comment]
@aurelienpierre indeed that a nice and informative video!

About the tone equalizer module i really love it although it needs some time to get it's potential.
Some points i am surprised about:
1. The default setting leads in many situations to an orange bar in "mask post-processing" . Ok,no problem; but as a  default, shouldn't it be more "conservative"?
2. If i scroll / shift through exposure compensation the "mask post-processing" jumps around and seems not to correspong to slight changes. Is this maths or a bug?
3. Shouldn't there also be an orange bar at the left?

**Comment by dcendal** [comment]
@aurelienpierre, first off all, thanks for the great work.
I see your video for a second time and now I'm not sure if I'm doing something wrong or not.
If you take a look to my 2nd and 3rd snapshots the  post-procesing mask is almost in the middle , the histogram is also in the middle and the curve to 0. Is there something more that I should consider to know that I'm introducing the parameters in a bad way?
Thanks again.

**Comment by cigno5** [comment]
@aurelienpierre let me say first that tone equalizer is amazing in many aspects and I'm very looking forward to master it.

but...
 
> Use the mask exposure compensation to slide and spread the histogram in the valid area [-8; 0] EV. If you pushed the exposure (in exposure module) by 2EV, you would need to pull it by -2EV in mask compensation.

I have a picture with overexposed area, I imported from scratch and I just enabled the tone equalizer module, producing the artifact. 

I don't want to whine (I know it's about tweaking it in the next weeks) but it seems counter-intuitive. Some commits ago it worked slightly different, and I felt it incredibly simple to use.
